neqo-transport/src/connection/mod.rs

```diff
@@ -2664,10 +2666,17 @@ impl Connection {
             let limit = if path.borrow().pmtud().needs_probe() {
                 needs_padding = true;
                 debug_assert!(path.borrow().pmtud().probe_size() >= profile.limit());
-                path.borrow().pmtud().probe_size() - aead_expansion
+                path.borrow().pmtud().probe_size()
+                    - if space == PacketNumberSpace::Initial {
+                        // Note that this will not reserve space for the indication
+                        // if packets are coalesced (with Handshake or 0-RTT).
+                        Self::SCONE_INDICATION.len()
```

Comment:
I don't yet understand the subtraction of `Self::SCONE_INDICATION.len()` here.

It reserves two extra bytes for scone, taking them away from `profile.limit()` which is calculated here.

But then below, we make sure we stay within the limit of `profile.limit()` when writing the SCONE bytes. 

``` rust
        let pad_amount = profile.limit() - encoder.len();
```

In other words, we never use the 2 reserved bytes. Am I missing something?